### PR TITLE
Add support for Windows Service Accounts

### LINKED ISSUE #1114: SCMU not compatible with service accounts, results in System.Exception: Not a valid account
I configured my ServiceControl instance to run under `NT Service\Particular.ServiceControl$SQL` but SCMU then fails to load with a **System.Exception: Not a valid account**

SCMU then quits, you cannot manage any instances.

```
System.Exception: Not a valid account
   at ServiceControlInstaller.Engine.Accounts.UserAccount.ParseAccountName(String accountName) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Accounts\UserAccount.cs:line 125
   at ServiceControlInstaller.Engine.Instances.ServiceControlInstance.DefaultLogPath() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Instances\ServiceControlInstance.cs:line 291
   at ServiceControlInstaller.Engine.Instances.ServiceControlInstance.Reload() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Instances\ServiceControlInstance.cs:line 57
   at ServiceControlInstaller.Engine.Instances.InstanceFinder.<>c.<ServiceControlInstances>b__2_1(WindowsServiceController p) in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Instances\Instances.cs:line 33
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at ServiceControlInstaller.Engine.Instances.InstanceFinder.ServiceControlInstances() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Instances\Instances.cs:line 33
   at ServiceControlInstaller.Engine.Instances.InstanceFinder.AllInstances() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControlInstaller.Engine\Instances\Instances.cs:line 50
   at ServiceControl.Config.UI.ListInstances.ListInstancesViewModel.RefreshInstances() in C:\BuildAgent\work\7189a56f9f44affc\src\ServiceControl.Config\UI\ListInstances\ListInstancesViewModel.cs:line 36
```

image

### PR DESCRIPTION
**Pull request by ramonsmits**
Added support for Windows Service Accounts to run least privileged, resolves #1114

### CODE REVIEW

**Comment by ramonsmits** [comment]
Works locally

servicecontrolconfig_2018-08-22_15-33-33

File: src/ServiceControlInstaller.Engine/Accounts/UserAccount.cs [hunk 1]
```diff
@@ -26,6 +26,11 @@ public bool IsLocalService()
             return SID.IsWellKnown(WellKnownSidType.LocalServiceSid);
         }
 
+        bool IsServiceAccount()
+        {
+            return Domain == "NT SERVICE";
```
[the remark this answers is not in the document]
Comment (ramonsmits) [reply src/ServiceControlInstaller.Engine/Accounts/UserAccount.cs]: A built-in account is not a service account as these are not linked 1-1 to a service. You could create a local account and use these credentials to run a service, but you could run multiple services with that same account. A service account is special in that you cannot start a process using these credentials hence why its much more secure.

Comment (WilliamBZA) [inline src/ServiceControlInstaller.Engine/Accounts/UserAccount.cs]: The `NT SERVICE` check in the `LocalizedNTAuthority` method uses `Domain.Equals("NT SERVICE", StringComparison.CurrentCulture)`  Does it matter that this is different? Could there be languages where it is not the same spelling?

**Comment by WilliamBZA** [comment]
@ramonsmits there's also a bunch of failing tests on various transports. I can't quite tell why they would be failing.

**Comment by WilliamBZA** [comment]
@Particular/servicecontrol-maintainers I don't know enough about service accounts to progress with this.

@ramonsmits 

> The NT SERVICE check in the LocalizedNTAuthority method uses Domain.Equals("NT SERVICE", StringComparison.CurrentCulture) Does it matter that this is different? Could there be languages where it is not the same spelling?

**Comment by SzymonPobiega** [comment]
@ramonsmits @WilliamBZA guys, can you finish this one or kill it?

**Comment by WilliamBZA** [comment]
@SzymonPobiega:

> @Particular/servicecontrol-maintainers I don't know enough about service accounts to progress with this.

I have reach out for help as I don't know what to do. No response though.

**Comment by SzymonPobiega** [comment]
@WilliamBZA should we close this for now then? It is not tied to any prioritized work and the fact it sits here makes it build every time develop changes.

**Pull request merged**